README.md

```diff
@@ -55,3 +55,76 @@ You can install them using the following command:
 sudo apt update
 sudo apt install -y libssl-dev pkg-config
 ```
+
+## Step-by-Step Guides
+
+### Prerequisites
+Ensure you have the following installed:
+ * Rust
+ * Cargo
+
+## Setup
+1. Create a working directory eg(cd didcomm-mediator-rs) and cd into your directory.
+
+ 2. Clone the repository using the following command:
+
+git clone https://github.com/adorsys/didcomm-mediator-rs.git
+
+## Configuration Options
+The following configuration options are available:
+* `--config-file`: Path to the configuration file. Defaults to `config.toml`.
+* `--config`: Configuration options in TOML format. Overrides the configuration file.
+* `--log-level`: Log level. Can be one of `trace`, `debug`, `
+info`, `warn`, `error`, or `off`. Defaults to `info`.
+
+## Troubleshooting Tips
+If you encounter any issues while running the application, here are some troubleshooting tips to help you resolve them
+ 
+ ### Common Issues
+
+  1. Build Errors:
+  * Ensure that you have the required system packages installed. `libssl-dev` and `pkg-config` 
+  * Ensure that you have the latest version of Rust and Cargo installed.
+  * CHeck for any missing dependencies using `cargo check`.
+
+  2. Runtime Errors:
+  * Ensure that the configuration file is valid and that all required fields are present.
+  * Ensure the mediator service is running and accessible.
+
+### Solutons
+ Check the missing dependencies 
+
+### Configuration Errors:
+ * Double-check configuration files for typos or missing entries.
+ * Refer to the crate-specific documentation for required configuration settings.
+
+## Examples
+### Sending a DIDComm Message
+ 1. Start the mediator service:
+
+cd didcomm-mediator-rs/mediator
+cargo run
+
+2. Send a message using curl:
+
+curl -X POST http://localhost:8000/messages \
+-H "Content-Type: application/json" \
+-d '{
+    "id": "1234567890",
+    "type": "https://didcomm.org/routing/2.0/forward",
+    "to": ["did:example:bob"],
+    "message": {
+        "@id": "abcdefg",
+        "@type": "https://didcomm.org/routing/2.0/forward",
+        "to": ["did:example:alice"],
+        "message": {
+            "@id": "123",
```

Comment:
This is good but should be in the general guide of the setup

Reply:
Sure I agree @Christiantyemele . But any proposed file where it could be?

Reply:
I think the 1. Start the mediator service:  
should stay but the 
2. Send a message using curl is the one to be displaced or removed